lib/select/select.tsx

```diff
@@ -0,0 +1,336 @@
+import _ from 'lodash';
+import React from 'react';
+
+import { NAMESPACE } from '../utilities/ts/constants';
+
+import BEM from '../utilities/ts/bem';
+import namespace from '../utilities/ts/namespace';
+
+import Icon from '../icon';
+
+interface Options {
+  label: string;
+  value: any;
+}
+
+interface Props {
+  className?: string;
+  disabled?: boolean;
+  id?: string;
+  label: string;
+  message?: string;
+  modifiers?: string;
+  name?: string;
+  onChange?: (value: string, name: string | number) => void;
```

Comment:
Can the name argument now just be of type `'string'` as we've updated `props.name` above.